extensions/amp-mega-menu/0.1/amp-mega-menu.js

```diff
@@ -0,0 +1,363 @@
+    heading.setAttribute('aria-haspopup', 'dialog');
+    heading.setAttribute('aria-controls', contentId);
+    if (!heading.hasAttribute('tabindex')) {
+      heading.setAttribute('tabindex', 0);
+    }
+    heading.setAttribute('aria-expanded', 'false');
+    heading.addEventListener('click', e => this.handleHeadingClick_(e));
+    // prevent focus on mousedown so that there's no sudden shift in focus
+    // when the content container opens.
+    heading.addEventListener('mousedown', e => e.preventDefault());
+    heading.addEventListener('keydown', e => this.handleHeadingKeyDown_(e));
+  }
+
+  /**
+   * Handler for item heading's on-click event to expand/collapse its content.
+   * @param {!Event} event click event.
+   * @private
+   */
+  handleHeadingClick_(event) {
+    event.preventDefault();
+    event.stopPropagation();
+    const item = dev().assertElement(event.target.parentElement);
+    const previousItem = this.collapse_();
+    if (item != previousItem) {
+      this.expand_(item);
+    }
+  }
+
+  /**
+   * Handler for key presses on an item heading.
+   * @param {!Event} event keydown event.
+   * @private
+   */
+  handleHeadingKeyDown_(event) {
+    if (event.defaultPrevented) {
+      return;
+    }
+    const {key} = event;
+    switch (key) {
+      case Keys.LEFT_ARROW: /* fallthrough */
+      case Keys.RIGHT_ARROW:
+        this.handleNavigationKeyDown_(event);
+        return;
+      case Keys.ENTER: /* fallthrough */
+      case Keys.SPACE:
+        if (event.target == event.currentTarget) {
+          this.handleHeadingClick_(event);
+        }
+        return;
+    }
+  }
+
+  /**
+   * Handler for arrow key presses to navigate between menu items.
+   * @param {!Event} event keydown event.
+   * @private
+   */
+  handleNavigationKeyDown_(event) {
+    const item = dev().assertElement(event.target.parentElement);
+    const index = this.items_.indexOf(item);
+    if (index !== -1) {
+      event.preventDefault();
+      let dir = event.key == Keys.LEFT_ARROW ? -1 : 1;
+      // Left is 'previous' in LTR and 'next' in RTL; vice versa for Right.
+      if (isRTL(this.document_)) {
+        dir = -dir;
+      }
+      // If user navigates one past the beginning or end, wrap around.
+      const newIndex = mod(index + dir, this.items_.length);
+      tryFocus(this.getItemHeading_(this.items_[newIndex]));
+    }
+  }
+
+  /**
+   * Expand the given item to show its menu content.
+   * @param {!Element} item
+   * @private
+   */
+  expand_(item) {
+    this.mutateElement(() => {
+      // Wait for mutateElement, so that the element has been transfered to the
+      // fixed layer. This is needed to hide the correct elements.
+      const content = this.getItemContent_(item);
+      setModalAsOpen(content);
+      content.setAttribute('aria-modal', 'true');
+    });
+    item.setAttribute('open', '');
+    this.element.setAttribute('open', '');
+    this.maskElement_.setAttribute('open', '');
```

Comment:
Nit (and not required to be changed!): Can be expressed as a [].forEach since the operation is the same per element.

`[item, this.element, this.maskElement_].forEach(e=>e.setAttribute('open', ''));`